From 71aab10b92eede51527e86b6c01abb21fe7d4f96 Mon Sep 17 00:00:00 2001
From: Miguel Gomez <magomez@igalia.com>
Date: Thu, 2 Apr 2020 09:39:30 +0200
Subject: [PATCH] Avoid crash with cairo 1.16

---
 .../graphics/freetype/FontCustomPlatformDataFreeType.cpp    | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Source/WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp b/Source/WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp
index 297d329de04a..aca5cd8115ed 100644
--- a/Source/WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp
+++ b/Source/WebCore/platform/graphics/freetype/FontCustomPlatformDataFreeType.cpp
@@ -95,6 +95,12 @@ static bool initializeFreeTypeLibrary(FT_Library& library)
     // https://www.freetype.org/freetype2/docs/design/design-4.html
     // https://lists.nongnu.org/archive/html/freetype-devel/2004-10/msg00022.html
 
+    int version = cairo_version();
+    if (version >= CAIRO_VERSION_ENCODE(1, 15, 0) && version < CAIRO_VERSION_ENCODE(1, 16, 1)) {
+        FT_Init_FreeType(&library);
+        return true;
+    }
+
     FT_Memory memory = bitwise_cast<FT_Memory>(ft_smalloc(sizeof(*memory)));
     if (!memory)
         return false;
